### PR TITLE
Clean and document C++ code

### PR DESCRIPTION
**Pull request by tnagler**

### CODE REVIEW

File: inst/include/lpdens.hpp [hunk 1]
```diff
@@ -5,231 +5,286 @@
...
-        return tools::invert_f(x,
-                               f,
-                               grid_.get_grid_points().minCoeff(),
-                               grid_.get_grid_points().maxCoeff(),
-                               20);
-    }
-
-    Eigen::VectorXd get_values() const {return values_;}
-    Eigen::VectorXd get_grid_points() const {return grid_points_;}
+    // gettters
```
Comment (tvatter) [inline inst/include/lpdens.hpp]: gettters => getters

File: inst/include/lpdens.hpp [hunk 2]
```diff
@@ -5,231 +5,286 @@
...
+                          double xmax) :
+    bw_(bw),
+    xmin_(xmin),
+    xmax_(xmax)
+{
+    if (bw <= 0)
+        throw std::runtime_error("bw must be positive.");
+
+    // remove NaNs from the data
+    x = tools::nan_omit(x);
```
Comment (tvatter) [inline inst/include/lpdens.hpp]: Aren't we doing this in R already?

File: inst/include/lpdens.hpp [hunk 3]
```diff
@@ -5,231 +5,286 @@
...
+
+        // otherwise calculate normal pdf
+        double val = stats::dnorm(Eigen::VectorXd::Constant(1, xx))(0);
+
+        return val / 0.999999426697;  // correct for truncation
+    };
+    return x.unaryExpr(f);
+}
+
+//! (analytitcally) evaluates the kernel density estimate and its influence
```
Comment (tvatter) [inline inst/include/lpdens.hpp]: analytically

File: inst/include/lpdens.hpp [hunk 4]
```diff
@@ -5,231 +5,286 @@
...
+inline Eigen::VectorXd LPDens1d::boundary_correct(const Eigen::VectorXd& x,
+                                                  const Eigen::VectorXd& fhat)
+{
+    Eigen::VectorXd corr_term(fhat.size());
+    if (!std::isnan(xmin_) & !std::isnan(xmax_)) {
+        // two boundaries -> probit transform
+        corr_term = (x.array() - xmin_ + 5e-3) / (xmax_ - xmin_ + 1e-2);
+        corr_term = stats::dnorm(stats::qnorm(corr_term));
+        corr_term /= (xmax_ - xmin_ + 1e-2);
+        corr_term = 1 / corr_term.array().max(1e-4);
```
Comment (tvatter) [inline inst/include/lpdens.hpp]: 1.0 instead of 1?

File: inst/include/lpdens.hpp (hunk 2 above)
Comment (tnagler) [reply inst/include/lpdens.hpp]: Yeah, right I added that later; will remove.

File: inst/include/lpdens.hpp [hunk 5]
```diff
@@ -5,231 +5,286 @@
...
+//! corrects the density estimate for a preceding boundary transformation of
+//! the data.
+//! @param x evaluation points (in original domain).
+//! @param fhat the density estimate evaluated in the transformed domain.
+//! @return corrected density estimates at `x`.
+inline Eigen::VectorXd LPDens1d::boundary_correct(const Eigen::VectorXd& x,
+                                                  const Eigen::VectorXd& fhat)
+{
+    Eigen::VectorXd corr_term(fhat.size());
+    if (!std::isnan(xmin_) & !std::isnan(xmax_)) {
```
Comment (tvatter) [inline inst/include/lpdens.hpp:194]: Didn't we have compatibility issues with the `std::isnan`? In `vinecopulib`, we do `(boost::math::isnan)(something)`.

File: inst/include/lpdens.hpp [hunk 6]
```diff
@@ -5,231 +5,286 @@
...
-            grid_points(0) = xmin_;
-        if (!std::isnan(xmax_))
-            grid_points(grid_points.size() - 1) = xmax_;
-        return grid_points;
+    return fhat.array() * corr_term.array();
+}
+
+//! constructs a grid that is later used for interpolation.
+//! @param x vector of observations.
+//! @return a grid of size 200.
```
Comment (tvatter) [inline inst/include/lpdens.hpp]: 50, and shouldn't it be an argument (set by default to 50)?

File: inst/include/lpdens.hpp (hunk 5 above)
Comment (tnagler) [reply inst/include/lpdens.hpp:194]: I do remember that, but don' remember why. I'd prefer to keep it simple as long there's no error.

File: inst/include/lpdens.hpp (hunk 6 above)
Comment (tnagler) [reply inst/include/lpdens.hpp]: fixed the typo, would not make it argument until we think it's useful.

File: inst/include/lpdens.hpp (hunk 3 above)
Comment (tnagler) [reply inst/include/lpdens.hpp]: fixed

File: inst/include/lpdens.hpp (hunk 1 above)
Comment (tnagler) [reply inst/include/lpdens.hpp]: fixed

File: inst/include/lpdens.hpp (hunk 4 above)
Comment (tnagler) [reply inst/include/lpdens.hpp]: fixed

File: inst/include/lpdens.hpp (hunk 5 above)
Comment (tvatter) [reply inst/include/lpdens.hpp:194]: OK, let's keep this in mind when we do the continuous integration then.

**Pull request merged**